docs/getting-started/webassembly-wasi.md

```diff
@@ -0,0 +1,340 @@
+    ├── wasi-clocks-0.2.3
+    │   └── package.wit
+    ├── wasi-http-0.2.3
+    │   └── package.wit
+    ├── wasi-io-0.2.3
+    │   └── package.wit
+    └── wasi-random-0.2.3
+        └── package.wit
+```
+
+[wkg]: https://github.com/bytecodealliance/wasm-pkg-tools
+[wasi-http]: https://github.com/WebAssembly/wasi-http
+
+## 3. Hello Wasm
+
+Let's fulfill our `component` world with a basic Hono application as a WebAssembly component:
+
+```ts
+// component.mts
+import { Hono } from 'hono'
+
+const app = new Hono();
+
+app.get('/hello', (c) => {
+  return c.json({
+    message: 'Hello WASM!',
+  })
+})
+
+/**
+ * Register the Hono application with the global fetch listener as supported by the underlying StarlingMonkey JS runtime.
+ *
+ * Since both Hono and StarlingMonkey are aligned Web Standards (WinterCG/WinterTC),
+ * this enables Hono to run smoothly in WASI-enabled (`wasi:http`) Webassembly environments.
+ *
+ * See: https://github.com/bytecodealliance/ComponentizeJS#using-starlingmonkeys-fetch-event
+ * See: https://hono.dev/docs/concepts/web-standard
+ * See: https://wintertc.org/
+ * See: https://github.com/WebAssembly/wasi-http
+ */
+app.fire();
+```
+
+## 4. Build
+
+Since we're using Rollup (and it's configured to handle Typescript compilation), we can use it to build and bundle:
+
+::: code-group
+
+```sh [npm]
+npx rollup -c
+```
+
+```sh [yarn]
+yarn rollup -c
+```
+
+```sh [pnpm]
+pnpx rollup -c
+```
+
+```sh [bun]
+bun build --target=bun --outfile=dist/component.js ./component.mts
+```
+
+:::
+
+::: info
+The bundling step is necessary because WebAssembly JS ecosystem tooling only currently supports a single JS file,
+and we'd like to include Hono along with related libraries.
+
+For components with simpler requirements, bundlers are not necessary.
+:::
+
+To build your WebAssembly component, use `jco` (and indirectly `componentize-js`):
+
+::: code-group
+
+```sh [npm]
+npx jco componentize -w wit -o dist/component.wasm dist/component.js
+```
+
+```sh [yarn]
+yarn jco componentize -w wit -o dist/component.wasm dist/component.js
+```
+
+```sh [pnpm]
+pnpx jco componentize -w wit -o dist/component.wasm dist/component.js
+```
+
+```sh [bun]
+bun run jco componentize -w wit -o dist/component.wasm dist/component.js
+```
+
+:::
+
+
+## 3. Run
+
+To run your Hono WebAssembly HTTP server, you can use any WASI-enabled WebAssembly runtime:
+
+- [`wasmtime`][wasmtime]
+- `jco` (runs in NodeJS)
+
+In this guide, we'll use `jco serve` since it's already installed.
+
+::: warning
+`jco serve` is meant for development, and is not recommended for production use.
+:::
+
+[wasmtime]: https://wasmtime.dev
+
+::: code-group
+
+```sh [npm]
+npx jco serve dist/component.wasm
+```
+
+```sh [yarn]
+yarn jco serve dist/component.wasm
+```
+
+```sh [pnpm]
+pnpx jco serve dist/component.wasm
+```
+
+```sh [bun]
+bun run jco serve dist/component.wasm
+```
+
+:::
+
+You should see output like the following:
+
+```
+$ npx jco serve dist/component.wasm
```

Comment:
Does `jco` really exist on npm now? It returns 404.

![CleanShot 2025-06-26 at 17 28 55@2x](https://github.com/user-attachments/assets/a4199e1d-c5df-4c42-9ad3-54ab4b087eb5)